src/design-system/color/useForegroundColor.ts

```diff
@@ -22,7 +23,9 @@ export function useForegroundColors(
 
   return colors.map(color => {
     if (color === 'accent') {
-      return accentColor.color;
+      return accentColor
+        ? accentColor.color
+        : getDefaultAccentColorForColorMode(colorMode).color;
```

Comment:
This is where we dynamically resolve the default accent color based on the color mode this component is in.